Source/Controller/MainViewController.swift

```diff
@@ -123,6 +116,26 @@ class MainViewController: UITabBarController {
     // custom separator on the top edge of the tab bar
     private var topBorder: UIView?
 
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        homeFeatureViewController = FeatureViewController(
+            rootViewController: UIHostingController(
+                rootView: HomeView()
+                    .environmentObject(BotRepository.shared)
+                    .environmentObject(helpDrawerState)
```

Comment:
I think `helpDrawerState` is being inserted too high up in the state hierarchy here. IMO we should reserve `.environmentObject()` for global singletons that are needed by many views deep in the hierarchy. In this case I think we could just pass `helpDrawerState` into `HomeView.init()` and keep things a little cleaner.

Reply:
👌